### PR TITLE
feat: add deal status update to endblocker

### LINKED ISSUE #414: Update a Deal Status in Endblocker
Update a Deal Status in `EndBlocker`.
- When the deal is ready or being started, the status should be `ACTIVE`.
- When the deal is completed, the status should be `COMPLETED`.

### PR DESCRIPTION
**Pull request by audtlr24**
Close #414 
Add deal status update to `DataVerficationEndBlocker` using `IsDealCompleted()` and `GetDealCurNumDataAndIncrement`.
- check `Deal` is completed or not, and increase related `Deal`'s `curNumData` if it is not completed.
- add `DATA_SALE_STATUS_DEAL_FULL` to distinguish `DATA_SALE_STATUS_VERIFICATION_FAILED` (Later, seller can re-request `sellData` when the status is `DATA_SALE_STATUS_VERIFICATION_FAILED`)

### CODE REVIEW

**Review by gyuguen — APPROVED** [review]
LGTM
I left an unImportented comment

File: x/datadeal/abci.go [hunk 1]
```diff
@@ -35,20 +35,29 @@ func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
...
-
+			isDealCompleted, err := keeper.IsDealCompleted(ctx, dataSale.DealId)
+			if err != nil {
+				panic(err)
+			}
+
+			if isDealCompleted {
+				dataSale.Status = types.DATA_SALE_STATUS_DEAL_FULL
+			} else {
+				if err = keeper.GetDealCurNumDataAndIncrement(ctx, dataSale.DealId); err != nil {
```
[the remark this answers is not in the document]
Comment (audtlr24) [reply x/datadeal/abci.go]: Thanks! I agree. I rename it in [ed9306a](https://github.com/medibloc/panacea-core/pull/470/commits/ed9306a85e7f39870d59148921e3560554b1537e).

File: proto/panacea/datadeal/v2alpha2/datasale.proto [hunk 2]
```diff
@@ -41,4 +41,7 @@ enum DataSaleStatus {
 
   // DATA_SALE_STATUS_DELIVERY_FAILED defines that the status of DataSale is delivery failed.
   DATA_SALE_STATUS_DELIVERY_FAILED = 5;
+
+  //DATA_SALE_STATUS_DEAL_FULL defines that the status of Deal associated with DataSale is full.
+  DATA_SALE_STATUS_DEAL_FULL = 6;
```
Comment (inchori) [inline proto/panacea/datadeal/v2alpha2/datasale.proto:46]: IMO, It doesn't matter. But I think that you can change this name to `DATA_SALE_STATUS_DEAL_COMPLETED`.

**Review by inchori — APPROVED** [review]
lgtm!
I just left a comment.

**Review by H4NLee — APPROVED** [review]
lgtm

I suggest to set deal in `abci_test.go` using `CreateDeal`, not `SetDeal` directly.
The reason why the `keeper.EndBlocker` do somethings deeply related to deal, vote, and datasale. 
For example, if we set the deal directly, the deal has no budget because the logic for budget is implemented in `CreateDeal`. Then `abci_test.go` will fail when reward-distribution is added.

File: proto/panacea/datadeal/v2alpha2/datasale.proto (hunk 2 above)
Comment (audtlr24) [reply proto/panacea/datadeal/v2alpha2/datasale.proto]: Okay, That seems to be a clearer expression. I change it.

**Comment by audtlr24** [comment]
> lgtm
> 
> I suggest to set deal in `abci_test.go` using `CreateDeal`, not `SetDeal` directly. The reason why the `keeper.EndBlocker` do somethings deeply related to deal, vote, and datasale. For example, if we set the deal directly, the deal has no budget because the logic for budget is implemented in `CreateDeal`. Then `abci_test.go` will fail when reward-distribution is added.

I agree, so I change `SetDeal` part to use `CreateDeal` in `abci_test.go`. Thanks!

**Pull request merged**